Sources/App/Core/Search.swift

```diff
@@ -186,6 +232,127 @@ enum Search {
+     "SimpleRoulette"    0.06079271    "'anim':19 'bezier':5 'carthag':6 'chart':7,8 'cocoapod':9 'compon':29 'creat':1 'ease.simplerouletteangle':4 'framework':32 'io':10,12 'ios-swift':11 'maco':14 'pod':15 'roulett':2,16,18,21 'roulette-anim':17 'roulette-wheel':20 'simpl':23 'spm':24 'swift':13,25 'swiftui':26,28,31 'swiftui-compon':27 'swiftui-framework':30 'wheel':22"
+     
+     -- same sample, except for 'wtv'
+     
+     "package_name"    "rank"    "combined_vector"
+     "WTV"    0.06079271    "'reflect':6 'swift':7 'variabl':4 'wtv':8 'wtvmirror':5"
+     "DataObject"    0.06079271    "'dataobject':1,4 'dataobjectani':3 'json':5 'object':6 'surl':7 'swift':8 'wtv':9"
+     
+     -- sample for 'ping'
+     
+     "package_name"    "rank"    "combined_vector"
+     "SwiftyPing"    0.06079271    "'5swiftypingicmp':7 '5swiftypingicmp-ping':6 'client':3 'icmp':1 'network':9 'ping':2,8 'swift':5 'swift4':10 'swift5':11"
+     
+     -- sample for guass-krueger
+     
+     "package_name"    "rank"    "combined_vector"
+     "GaussKrueger"    0.26832977    "'convert':1 'coordin':3 'gauss':10 'gauss-krueg':9 'gk4':2 'krueger':11 'versagausskruegercoordin':8 'vice':7 'wgs84':5,12"
+     
+     -- sample for iso639-1
+     
+     "package_name"    "rank"    "combined_vector"
+     "ISO639"    0.26832977    "'-1':7 '-2':9 'iso639':1,6,8 'iso639iso639':5 'languag':2,10,11 'librari':15 'manag':19 'packag':18 'swift':4,12,14,17 'swift-librari':13 'swift-package-manag':16 'swift5':20"
+     
+     -- sample for '具方法'
+     - zero results
+     
+     -- create a combined ts_vector against which to compare
+     to_tsvector(coalesce(summary, '') || coalesce(package_name, '') || coalesce(ARRAY_TO_STRING(keywords, ' '), '')),
+
+     
+     -- I think we want to use ts_rank INSTEAD of ts_rank_cd because most of the search queries are single
+     -- terms, and the cover distance value (returned as a rank parameter) doesn't help at all in single-word queries.
+     
+     -- ts_rank _can_ take a normalization parameter - and integer that reflects how the ranking values are normalized.
+     -- 32 (used in the ts_rank_cd example below) indicates: rank/(rank+1), which forces the ranking between 0 and 1.
+     -- the normalization factor is *always* related to the document length, which in our case isn't very useful since
+     -- all the documents are roughly the same length. In the case of ts_rank_cd, it can also be related to the distance
+     -- between search values - the distances between them in tsvector.
+     
+     -- On the plus side, we can get a rank for a non-matching term that reliably comes back as 0, so we can
+     -- work with that to provide additional tweaking. Example:
+     
+     SELECT query, a_vector, ts_rank(a_vector, query) AS rank
+     FROM plainto_tsquery('fiddlesticks') query,
+          to_tsvector('Heres my example package of swift package manager') a_vector
+--      WHERE query @@ a_vector
+     ORDER BY rank DESC;
```

Comment:
Should we schedule a call to go through this at some point? There's a lot of interesting stuff here, and it might be best to go through this together in a quick back and forth rather than an extended async PR discussion?

Reply:
Sounds excellent to me! When you're not slammed, let me know and I can step through things. I'd love the feedback on "proper" SQLKit structure in case I've chosen incorrectly, and in the mean time I've a few pieces to iterate out (the SQLKit for concatenating the tsvectors with weights, getting tests operational, verifying the performance characteristics)